### PR TITLE
70 bug login screen has back button

### PR DESCRIPTION
**Pull request by VVMaxVV**
Fixed bug with back button on AuthFragment
https://trello.com/c/KDrJnb4M/70-buglogin-screen-has-back-button

### CODE REVIEW

File: app/src/main/java/com/example/internetshop/data/cache/TokenPreference.kt [hunk 1]
```diff
@@ -16,6 +16,7 @@ class TokenPreference @Inject constructor(context: Context) {
         context.getSharedPreferences(PREFERENCE_NAME, Context.MODE_PRIVATE)
 
     fun getToken(): Token {
+        setToken(Token(null,Date()))
```
Comment (darthoo) [inline app/src/main/java/com/example/internetshop/data/cache/TokenPreference.kt:19]: What is it? The first line of getToken is setToken. Everything is correct here?

File: app/src/main/java/com/example/internetshop/presentation/activity/MainActivity.kt [hunk 2]
```diff
@@ -89,24 +92,24 @@ class MainActivity : AppCompatActivity(), ContainerHolder {
     private fun setupAppBar(navController: NavController) {
         WindowCompat.setDecorFitsSystemWindows(window, false)
         setSupportActionBar(binding?.toolbar)
+        val topLevelDestinations = setOf(
```
Comment (darthoo) [inline app/src/main/java/com/example/internetshop/presentation/activity/MainActivity.kt:95]: Define this set only in toolbar view model. And in app bar config use `toolbarViewModel.topLevelDestinations`

**Review by darthoo — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/internetshop/data/cache/TokenPreference.kt (hunk 1 above)
Comment (VVMaxVV) [reply app/src/main/java/com/example/internetshop/data/cache/TokenPreference.kt:19]: fixed

File: app/src/main/java/com/example/internetshop/presentation/activity/MainActivity.kt (hunk 2 above)
Comment (VVMaxVV) [reply app/src/main/java/com/example/internetshop/presentation/activity/MainActivity.kt]: fixed

**Review by darthoo — APPROVED** [review]

**Pull request merged**